ai/agents.py

```diff
@@ -77,9 +80,11 @@ def on_pick_start(self) -> Tuple[float, float]:
         """ Must return a valid starting position for the agent """
         return (random.random() * self.map.width, random.random() * self.map.height)
 
-    def on_noise(self, noise: world.NoiseEvent) -> None:
+    def on_noise(self, noises: List['world.NoiseEvent']) -> None:
         """ Noise handler, will be called before `on_tick` """
-        ...
+        angle = super().on_noise(noises);
+        if(angle != None):
+            self.heading = (angle+180)%360
```

Comment:
You shouldn't set the heading directly, that bypasses all the movement code and basically allows instant movement. But this is probably my fault for making it public. My bad.

Reply:
I agree. The agent should turn properly